matrix-sdk/src/main/java/org/matrix/androidsdk/data/Room.java

```diff
@@ -1971,26 +1969,13 @@ private void handleRoomAccountDataEvents(List<Event> accountDataEvents) {
                         }
                     }
                 } else {
-                    mAccountData.handleTagEvent(accountDataEvent);
+                    mAccountData.handleEvent(accountDataEvent);
                     if (Event.EVENT_TYPE_TAGS.equals(accountDataEvent.getType())) {
                         if (summary != null) {
-                            summary.setRoomTags(mAccountData.getKeys());
+                            summary.setRoomTags(mAccountData.getRoomTagsKeys());
                             getStore().flushSummary(summary);
```

Comment:
maybe you could flush a t the end of the for

Reply:
I agree, I will merge first PR #531 on this branch then apply your suggestion

Reply:
@BillCarsonFr  don't you think this flush was required before calling `mDataHandler` listeners (see `.onRoomTagEvent` and `onReadMarkerEvent`)?
I hesitate to move this flush. wdyt?

Reply:
yes, keep it like that. better not break anything